### PR TITLE
QM : Report value from generic one-time-checkout

### PR DESCRIPTION
**Pull request by paul-daniel-dempsey**
## What are you doing in this PR?

Objective: Report generic oneTimeCheckout value from generic checkout on page load.

This PR does not cover:
- Reporting one-time-checkout value from the 3-tier landing page
- Firing the new conversion events from the generic one-time-checkout.

These can be tackled in follow-up PRs.

**DONE:**

- Create `sendEventCheckoutValue` function [DONE]
- Call `sendEventCheckoutValue` function on generic oneTime checkout Tier 1/2/3 page load [DONE]

### CODE REVIEW

**Comment by GHaberis** [comment]
We should merge this PR first https://github.com/guardian/support-frontend/pull/6445 before reviewing this

File: support-frontend/assets/helpers/tracking/quantumMetric.ts [hunk 1]
```diff
@@ -125,12 +126,11 @@ function sendEvent(
...
-			id,
-			qmCartValueEventId,
-			value,
-			payload ?? {},
-		);
+		if (payload) {
+			window.QuantumMetricAPI.sendEvent(id, qmCartValueEventId, value, payload);
+		} else {
+			window.QuantumMetricAPI.sendEvent(id, qmCartValueEventId, value);
+		}
```
Comment (GHaberis) [inline support-frontend/assets/helpers/tracking/quantumMetric.ts:133]: Can we replace the `if else` with:

`window.QuantumMetricAPI.sendEvent(id, qmCartValueEventId, value, payload ?? {});`

I made the change in my PR but I think it's been replaced in the rebase.

Comment (paul-daniel-dempsey) [reply support-frontend/assets/helpers/tracking/quantumMetric.ts]: sorry corrected now

File: support-frontend/assets/helpers/tracking/quantumMetric.ts [hunk 2]
```diff
@@ -298,6 +298,33 @@ function sendEventSubscriptionCheckoutConversion(
 	}
 }
 
+function sendEventOneTimeCheckoutValue(
+	amount: number,
+	sourceCurrency: IsoCurrency,
+): void {
+	console.log('*** sendEventOneTimeCheckoutValue ***', amount, sourceCurrency);
```
[the remark this answers is not in the document]
Comment (paul-daniel-dempsey) [reply support-frontend/assets/helpers/tracking/quantumMetric.ts]: missed that one too, removed

**Review by GHaberis — APPROVED** [review]
Nice 👍

**Pull request merged**